### PR TITLE
add fftshift to jax frontend

### LINKED ISSUE #20853: fftshift
This subtask is to add the `fftshift` function to the Jax frontend

### PR DESCRIPTION
**Pull request by hnhparitosh**
Greetings!
Added fftshift to jax frontend. Tests seem to be passing. Please review the PR and let me know if any changes are needed.
🙂
This PR will close #20853

### CODE REVIEW

**Comment by aparajith21** [comment]
Thanks for contributing, will check this soon!

File: ivy/functional/frontends/jax/numpy/fft.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+def fftshift(x, axes=None, name=None):
+    shape = x.shape
+
+    if axes is None:
+        axes = tuple(range(x.ndim))
+        shifts = [(dim // 2) for dim in shape]
+    elif isinstance(axes, int):
+        shifts = shape[axes] // 2
+    else:
+        shifts = ivy.concat([shape[ax] // 2 for ax in axes])
```
Comment (aparajith21) [inline ivy/functional/frontends/jax/numpy/fft.py:15]: Why is `ivy.concat` necessary here?

**Comment by aparajith21** [comment]
Could you please comment `add_frontend_checklist` and tick off the items there!

File: ivy/functional/frontends/jax/numpy/fft.py (hunk 1 above)
Comment (hnhparitosh) [reply ivy/functional/frontends/jax/numpy/fft.py:15]: My bad. I just wanted to concatenate the shift values in single array.
We can directly pass it to `ivy.roll` by omitting `ivy.concat`. It works the same.

**Comment by hnhparitosh** [comment]
add_frontend_checklist

**Comment by aparajith21** [comment]
Sorry for the delayed review, LGTM with the change, and once you resolve the conflicts, there shouldn't be other issues and the tests would pass.

**Comment by hnhparitosh** [comment]
Hi! Thank you for the feedback. I have made the necessary changes. I hope this is ready to merge.

File: ivy_tests/test_ivy/test_frontends/test_jax/test_numpy/test_fft.py [hunk 2]
```diff
@@ -46,3 +46,27 @@ def test_jax_numpy_fft(
...
+
+
+# fftshift
+@handle_frontend_test(
+    fn_tree="jax.numpy.fft.fftshift",
+    dtype_and_x=helpers.dtype_and_values(
+        available_dtypes=helpers.get_dtypes("valid"), shape=(4,), array_api_dtypes=True
+    ),
+)
+def test_jax_numpy_fttshift(
```
Comment (aparajith21) [inline ivy_tests/test_ivy/test_frontends/test_jax/test_numpy/test_fft.py:58]: `def test_jax_numpy_fftshift` minor typo fix please

**Comment by hnhparitosh** [comment]
Hi @aparajith21. I feel this should do it.

**Comment by aparajith21** [comment]
LGTM, thanks for contributing!

**Pull request merged**